tools/liveimage-mount

```diff
@@ -63,43 +79,74 @@ def call(*popenargs, **kwargs):
     return rc
 
 
-def rcall(args, env=None):
+def rcall(args, stdin=None, raise_err=True, env=None):
+    """Return stdout, stderr, & returncode from a subprocess call."""
+
+    out, err, p, environ = '', '', None, None
     if env:
         environ = os.environ.copy()
         environ.update(env)
-    else:
-        environ = None
     try:
-        p = subprocess.Popen(args, stdout=subprocess.PIPE,
-                             stderr=subprocess.PIPE, env=environ)
-        out, err = p.communicate()
-    except OSError, e:
-        raise CreatorError(u"Failed to execute:\n'%s'\n'%s'" % (args, e))
-    except:
-        raise CreatorError(u"""Failed to execute:\n'%s'
-            \renviron: '%s'\nstdout: '%s'\nstderr: '%s'\nreturncode: '%s'""" %
-            (args, environ, out, err, p.returncode))
+        p = subprocess.Popen(args, stdin=subprocess.PIPE, env=environ,
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, err = p.communicate(stdin)
+    except OSError as e:
+        out, err = out, u'Failed executing:\n%s\nerror: %s' % (args, e)
+        if raise_err:
+            raise LiveImageMountError(u'Failed executing:\n%s\nerror: %s' %
+                                      (args, e))
+    except Exception as e:
+        out, err = (out, u'Failed to execute:\n%s\nerror: %s' %
+                          (args, e))
+        if raise_err:
+            raise LiveImageMountError(u'''Failed to execute:\n%s
```

Comment:
We know how ``'''`` work, but since you use quite a lot of ``\n``s in the triple-quoted string anyway, I think it would just be less confusing to add some more at the ends of lines and use ``"string"`` ``"another string"`` (which Python concatenates as long as those are enclosed in parentheses) as @bcl suggests. You would also be able to get rid of the ``\r`` below since there will be no extra/carried-over identation.